proto/BUILD.bazel

```diff
@@ -19,133 +14,36 @@ license(
-    deps = [
-        ":p4types_proto",
-        "@com_google_protobuf//:any_proto",
-        "@com_google_protobuf//:descriptor_proto",
-    ],
-    # TODO(github.com/grpc/grpc/issues/20675): strip_import_prefix breaks
-    # cc_grpc_library. Make proto folder the Bazel root folder as a workaround.
-    # strip_import_prefix = "proto",
-)
+alias(name = "p4types_py_proto", actual = "//proto/p4/config/v1:p4types_py_proto")
+alias(name = "p4info_py_proto", actual = "//proto/p4/config/v1:p4info_py_proto")
+alias(name = "p4data_py_proto", actual = "//proto/p4/v1:p4data_py_proto")
+alias(name = "p4runtime_py_proto", actual = "//proto/p4/v1:p4runtime_py_proto")
 
-proto_library(
-    name = "p4runtime_proto",
-    srcs = ["p4/v1/p4runtime.proto"],
-    deps = [
-        ":p4data_proto",
-        ":p4info_proto",
-        "@com_google_googleapis//google/rpc:status_proto",
-        "@com_google_protobuf//:any_proto",
-    ],
-    # TODO(github.com/grpc/grpc/issues/20675): strip_import_prefix brakes
-    # cc_grpc_library. Make proto folder the Bazel root folder as a workaround.
-    # strip_import_prefix = "proto",
-)
-
-cc_proto_library(
-    name = "p4types_cc_proto",
-    deps = [":p4types_proto"],
-)
-
-cc_proto_library(
-    name = "p4info_cc_proto",
-    deps = [":p4info_proto"],
-)
-
-cc_proto_library(
-    name = "p4data_cc_proto",
-    deps = [":p4data_proto"],
-)
-
-cc_proto_library(
-    name = "p4runtime_cc_proto",
-    deps = [":p4runtime_proto"],
-)
-
-py_proto_library(
-    name = "p4types_py_proto",
-    deps = [":p4types_proto"],
-)
+alias(name = "p4info_go_proto", actual = "//proto/p4/config/v1:p4info_go_proto")
+alias(name = "p4runtime_go_proto", actual = "//proto/p4/v1:p4runtime_go_proto")
 
-py_proto_library(
-    name = "p4info_py_proto",
-    deps = [":p4info_proto"],
-)
-
-py_proto_library(
-    name = "p4data_py_proto",
-    deps = [":p4data_proto"],
-)
-
-py_proto_library(
-    name = "p4runtime_py_proto",
-    deps = [":p4runtime_proto"],
-)
-
-go_proto_library(
-    name = "p4info_go_proto",
-    importpath = "github.com/p4lang/p4runtime/go/p4/config/v1",
-    protos = [
-        ":p4info_proto",
-        ":p4types_proto",
-    ],
-)
-
-go_proto_library(
-    name = "p4runtime_go_proto",
-    importpath = "github.com/p4lang/p4runtime/go/p4/v1",
-    protos = [
-        ":p4data_proto",
-        ":p4runtime_proto",
-    ],
-    deps = [
-        ":p4info_go_proto",
-        "@com_google_googleapis//google/rpc:status_go_proto",
-    ],
-)
-
-cc_grpc_library(
-    name = "p4runtime_cc_grpc",
-    srcs = [":p4runtime_proto"],
-    generate_mocks = True,
-    grpc_only = True,
-    deps = [":p4runtime_cc_proto"],
-)
-
-py_grpc_library(
-    name = "p4runtime_py_grpc",
-    srcs = [":p4runtime_proto"],
-    deps = [":p4runtime_py_proto"],
-)
+alias(name = "p4runtime_cc_grpc", actual = "//proto/p4/v1:p4runtime_cc_grpc")
+alias(name = "p4runtime_py_grpc", actual = "//proto/p4/v1:p4runtime_py_grpc")
 
 build_test(
```

Comment:
Tests should live adjacent to the source files they test. Please break up into two tests in proto/p4/v1 and proto/p4/config/v1.